bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/connector/NonBlockingCallback.java

```diff
@@ -38,6 +38,7 @@ public NonBlockingCallback(Strand strand) {
         strand.setState(State.BLOCK_AND_YIELD);
         this.strand = strand;
         this.scheduler = strand.scheduler;
+        this.strand.setReturnValues(null);
```

Comment:
Why do we need to set null here? Can we set necessary return value in Kafka actions level?

Reply:
Callback should set the return values to null, when instantiating. Otherwise this may lead to errors.
Instead of setting this in every place where we create a callback, it's better to have it here, right?
@vinok88

Reply:
Yeah, this is not a Kafka specific thing. If you do not set return value properly, return values of previous nonblocking external calls will exist. This is a kind of a cleanup to avoid unintended results.

Reply:
Then why we faced this issue now. All these days we didn't face this issue without this?

Reply:
You won't get this if you set return values in each and every return points from java side (even if you are returning null). We enforce it here by setting it in the constructor itself.